samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -354,15 +358,29 @@ public void stop() {
     LOG.info("Stopping event hub system consumer...");
     List<CompletableFuture<Void>> futures = new ArrayList<>();
     streamPartitionReceivers.values().forEach((receiver) -> futures.add(receiver.close()));
-    CompletableFuture<Void> future = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
+    CompletableFuture<Void> allFuture = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
     try {
-      future.get(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
-    } catch (ExecutionException | InterruptedException | TimeoutException e) {
+      allFuture.get(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    } catch (Exception e) {
       LOG.warn("Failed to close receivers", e);
+      Util.logThreadDump("Log thread dump as we failed during consumer shutdown");
```

Comment:
the string is meant to be used as an id: maybe name it as - "Thread dump for EventHubSystemConsumer.PartitionReceiver#close failure"

Reply:
sure. fixing it